filament/src/main/java/net/fabricmc/filament/task/UnpickJarTask.java

```diff
@@ -56,25 +78,102 @@ public interface UnpickParameters extends WorkParameters {
 	public abstract static class UnpickAction implements WorkAction<UnpickParameters> {
 		@Override
 		public void execute() {
-			List<String> args = new ArrayList<>();
-			args.add(getPath(getParameters().getInput()));
-			args.add(getPath(getParameters().getOutput()));
-			args.add(getPath(getParameters().getUnpickDefinition()));
-			args.add(getPath(getParameters().getConstantJar()));
-
-			for (File file : getParameters().getClasspath().getFiles()) {
-				args.add(file.getAbsolutePath());
-			}
+			File outputFile = getParameters().getOutput().get().getAsFile();
 
 			try {
-				Main.main(args.toArray(String[]::new));
+				FileUtil.deleteIfExists(outputFile);
 			} catch (IOException e) {
 				throw new RuntimeException(e);
 			}
+
+			List<ZipFile> classpathZips = new ArrayList<>();
+
+			try (
+					ZipFile inputZip = new ZipFile(getParameters().getInput().get().getAsFile());
```

Comment:
This is fine for yarn, but in loom we tend to always use NIO zip fs via tiny remappers `FileSystemReference.openJar` (loom has loads of helpers around this) instead of the "legacy" ZipFile apis. Dont rewrite this now, but keep in mind for the future.

It may make more sense to try and share some of this code with loom, such as `ZipFsClasspath` and `AsyncZipProcessor`. It will likely need some loom changes to make it a better library.